### PR TITLE
Vulkan: Simple blacklist of device names that are not allowed to check for Vulkan

### PR DESCRIPTION
**Pull request by hrydgard**
Will help #12093.

Sad stuff..

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
I feel gross even suggesting this, but maybe catching the exception in the offending function would be "less dirty"?

File: Common/Vulkan/VulkanLoader.cpp [hunk 1]
```diff
@@ -239,6 +246,16 @@ void VulkanSetAvailable(bool available) {
 bool VulkanMayBeAvailable() {
 	if (g_vulkanAvailabilityChecked)
 		return g_vulkanMayBeAvailable;
+
+	std::string name = System_GetProperty(SYSPROP_NAME);
+	for (size_t i = 0; i < ARRAY_SIZE(device_name_blacklist); i++) {
```
Comment (CookiePLMonster) [inline Common/Vulkan/VulkanLoader.cpp]: Range based for loop would be a bit cleaner here, I think. Yay for not using macros.

**Comment by hrydgard** [comment]
I'm not sure how we could reliably recover from a crash caused by the detection.

Switched to range based for loop, I always forget those work on plain arrays.

**Review by unknownbrackets — APPROVED** [review]
I'm concerned that there may eventually be a new SHIELD Tablet with exactly the same name but without this problem, but not sure what to do about that, really...

-[Unknown]

**Pull request merged**